zerver/views/compatibility.py

```diff
@@ -10,3 +10,9 @@ def check_global_compatibility(request: HttpRequest) -> HttpResponse:
     if user_agent['name'] == "ZulipInvalid":
         return json_error("Client is too old")
     return json_success()
+
+def check_server_compatibility(request: HttpRequest) -> HttpResponse:
+    user_agent = parse_user_agent(request.META["HTTP_USER_AGENT"])
+    if user_agent['name'] == "ZulipInvalid":
+        return json_error("Client is too old")
+    return json_success()
```

Comment:
Do we want this as a separate endpoint, or would it make sense to just do whatever we're doing inside `/server_settings`?

Reply:
Hm, yeah, I think it would make sense to move this into the same object returned by `/server_settings`.

Reply:
I waffled on that question. I suggested a separate endpoint mainly because we don't currently check server_settings after the user first logs in.

We could move it inside; it'd just mean that this feature blocks on starting to routinely check server_settings, as we plan to do in zulip/zulip-mobile#2893.